flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NettyShuffleEnvironmentConfiguration.java

```diff
@@ -378,7 +378,9 @@ public static NettyShuffleEnvironmentConfiguration fromConfiguration(
         boolean batchShuffleCompressionEnabled =
                 configuration.get(NettyShuffleEnvironmentOptions.BATCH_SHUFFLE_COMPRESSION_ENABLED);
         String compressionCodec =
-                configuration.get(NettyShuffleEnvironmentOptions.SHUFFLE_COMPRESSION_CODEC);
+                configuration
+                        .get(NettyShuffleEnvironmentOptions.SHUFFLE_COMPRESSION_CODEC)
+                        .toString();
```

Comment:
Why would we want to convert this enum to a string. Looking into all its usages, eventually the string will be passed into `BlockCompressionFactory#createBlockCompressionFactory` and convert to `CompressionFactoryName`.